integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java

```diff
@@ -103,148 +137,392 @@
-    assertTrue(patchDomainCustomResource(domainUid, domainNamespace, patch, V1Patch.PATCH_FORMAT_JSON_PATCH),
-        "patchDomainCustomResource failed");
 
-    //verify domain changed event is logged
-    withStandardRetryPolicy
-        .conditionEvaluationListener(
-            condition -> logger.info("Waiting for domain event {0} to be logged "
-                + "(elapsed time {1}ms, remaining time {2}ms)",
-                DOMAIN_CHANGED,
-                condition.getElapsedTimeInMS(),
-                condition.getRemainingTimeInMS()))
-        .until(checkDomainEvent(opNamespace, domainNamespace, domainUid,
-            DOMAIN_CHANGED,  "Normal", timestamp));
+  /**
+   * Test DomainDeleted event is logged when domain resource is deleted.
+   */
+  @Order(9)
+  @Test
+  @DisplayName("Test domain events for various domain life cycle changes")
+  public void testDomainK8SEventsDelete() {
+    DateTime timestamp = new DateTime(Instant.now().getEpochSecond() * 1000L);
 
-    //verify domain processing retrying event
-    withStandardRetryPolicy
-        .conditionEvaluationListener(
-            condition -> logger.info("Waiting for domain event {0} to be logged "
-                + "(elapsed time {1}ms, remaining time {2}ms)",
-                DOMAIN_PROCESSING_RETRYING,
-                condition.getElapsedTimeInMS(),
-                condition.getRemainingTimeInMS()))
-        .until(checkDomainEvent(opNamespace, domainNamespace, domainUid,
-            DOMAIN_PROCESSING_RETRYING, "Normal", timestamp));
+    deleteDomainCustomResource(domainUid, domainNamespace1);
+    checkPodDoesNotExist(adminServerPodName, domainUid, domainNamespace1);
+    checkPodDoesNotExist(managedServerPodNamePrefix + 1, domainUid, domainNamespace1);
+    checkPodDoesNotExist(managedServerPodNamePrefix + 2, domainUid, domainNamespace1);
 
-    //verify domain processing aborted event
-    withStandardRetryPolicy
-        .conditionEvaluationListener(
-            condition -> logger.info("Waiting for domain event {0} to be logged "
-                + "(elapsed time {1}ms, remaining time {2}ms)",
-                DOMAIN_PROCESSING_ABORTED,
-                condition.getElapsedTimeInMS(),
-                condition.getRemainingTimeInMS()))
-        .until(checkDomainEvent(opNamespace, domainNamespace, domainUid,
-            DOMAIN_PROCESSING_ABORTED, "Warning", timestamp));
+    //verify domain deleted event
+    checkEvent(opNamespace, domainNamespace1, domainUid, DOMAIN_DELETED, "Normal", timestamp);
+  }
 
-    deleteDomainCustomResource(domainUid, domainNamespace);
-    checkPodDoesNotExist(adminServerPodName, domainUid, domainNamespace);
-    checkPodDoesNotExist(managedServerPodNamePrefix + 1, domainUid, domainNamespace);
-    checkPodDoesNotExist(managedServerPodNamePrefix + 2, domainUid, domainNamespace);
+  /**
+   * Test verifies NamespaceWatchingStarted event is logged when operator starts watching an another domain namespace.
```

Comment:
Need more description for the usecase. 
Usage of method 
   upgradeAndVerifyOperator(opNamespace, domainNamespace1, domainNamespace2);
looks like a blackbox testing. 
Can you embed the code here to provide a clear flow of the usecase.

Reply:
done